src/subdomains/supporting/notification/factories/mail.factory.ts

```diff
@@ -161,4 +210,57 @@ export class MailFactory {
 
     return { salutation, body, subject };
   }
+
+  private tNew(key: string, lang: string, args?: any): string {
+    return this.i18n.translate(key, { lang, args });
+  }
+
+  //*** MAIL BUILDING METHODS ***//
+
+  private getTable(table: Record<string, string>, lang: string): UserMailTable[] {
+    Util.removeNullFields(table);
+    return Object.entries(table).map((element) => ({
+      key: this.tNew(element[0], lang),
+      value: element[1],
+    }));
+  }
+
+  private getSuffix(suffix: TranslationItem[], lang: string): UserMailSuffix[] {
+    Util.removeNullFields(suffix);
+    return suffix.map((element) => this.mapSuffix(element, lang).flat()).flat();
+  }
+
+  private mapSuffix(element: TranslationItem, lang: string): MailSuffixMap[] {
+    switch (element.key) {
+      case MailKey.SPACE:
+        return [{ key: '', style: `${MailDefaultStyle};padding:1px` }];
+
+      case MailKey.DFX_TEAM_CLOSING:
+        return [
+          { key: '', style: `${MailDefaultStyle};padding:1px` },
+          { key: '', style: `${MailDefaultStyle};padding:1px` },
+          {
+            key: this.tNew(`${MailTranslationKey.GENERAL}.dfx_team_closing`, lang),
+            style: MailDefaultStyle,
+          },
+          { key: '', style: `${MailDefaultStyle};padding:1px` },
+          { key: '', style: `${MailDefaultStyle};padding:1px` },
+          { key: '', style: `${MailDefaultStyle};padding:1px` },
+          { key: '', style: `${MailDefaultStyle};padding:1px` },
+          { key: this.tNew(`${MailTranslationKey.GENERAL}.dfx_closing_message`, lang), style: 'Zapfino' },
+        ];
+
+      default:
+        return [
+          {
+            url: element.params?.url && {
+              link: element.params.url,
+              text: this.tNew(element.key, lang)?.split('[url:')[1]?.split(']')[0],
+            },
+            style: element.params?.style ?? MailDefaultStyle,
+            key: element.params?.url ? this.tNew(element.key, lang)?.split('[url:')[0] : this.tNew(element.key, lang),
+          },
+        ];
```

Comment:
Maybe do the splitting with regex and in a general method?
Parser method:
```ts
private parseSpecialTag(text: string): {text: string, tag: string, value: string} | undefined {
   const match = /^(.*)\[(\w*):(\w*)\]$/.exec(text);
   return match
      ? { text: match[1], tag: match[2], value: match[3] }
      : undefined;
}
```
Usage:
```ts
default:
   const text = this.tNew(element.key, lang);
   const specialTag = this.parseSpecialTag(text);

   return {
      url: specialTag?.tag === 'url' && element.params?.url ? {
         link: element.params.url,
         text: specialTag.value,
      } : undefined,
      style: element.params?.style ?? MailDefaultStyle,
      key: specialTag?.text ?? text;
   };
```

Reply:
Maybe you could also add an enum for special tags ...